sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Metrics/ExecutableLinesMetric.cs

```diff
@@ -0,0 +1,173 @@
+                    case SyntaxKind.ForStatement:
+                    case SyntaxKind.WhileStatement:
+
+                    case SyntaxKind.IfStatement:
+                    case SyntaxKind.LabeledStatement:
+                    case SyntaxKind.SwitchStatement:
+                    case SyntaxKind.ConditionalAccessExpression:
+                    case SyntaxKind.ConditionalExpression:
+
+                    case SyntaxKind.GotoStatement:
+                    case SyntaxKind.ThrowStatement:
+                    case SyntaxKind.ReturnStatement:
+                    case SyntaxKind.BreakStatement:
+                    case SyntaxKind.ContinueStatement:
+
+                    case SyntaxKind.YieldBreakStatement:
+                    case SyntaxKind.YieldReturnStatement:
+
+                    case SyntaxKind.SimpleMemberAccessExpression:
+                    case SyntaxKind.InvocationExpression:
+
+                    case SyntaxKind.SimpleLambdaExpression:
+                    case SyntaxKind.ParenthesizedLambdaExpression:
+
+                    case SyntaxKind.ArrayInitializerExpression:
+                        this.executableLineNumbers.Add(node.GetLocation().GetLineNumberToReport());
+                        return true;
+
+                    case SyntaxKind.StructDeclaration:
+                    case SyntaxKind.ClassDeclaration:
+                        return !HasExcludedCodeAttribute((BaseTypeDeclarationSyntax)node, btdc => btdc.AttributeLists,
+                            canBePartial: true);
+
+                    case SyntaxKind.MethodDeclaration:
+                    case SyntaxKind.ConstructorDeclaration:
+                        return !HasExcludedCodeAttribute((BaseMethodDeclarationSyntax)node, bmds => bmds.AttributeLists,
+                            canBePartial: true);
+
+                    case SyntaxKind.PropertyDeclaration:
+                    case SyntaxKind.EventDeclaration:
+                        return !HasExcludedCodeAttribute((BasePropertyDeclarationSyntax)node, bpds => bpds.AttributeLists);
+
+                    case SyntaxKind.AddAccessorDeclaration:
+                    case SyntaxKind.RemoveAccessorDeclaration:
+                    case SyntaxKind.SetAccessorDeclaration:
+                    case SyntaxKind.GetAccessorDeclaration:
+                        return !HasExcludedCodeAttribute((AccessorDeclarationSyntax)node, ads => ads.AttributeLists);
+
+                    default:
+                        return true;
+                }
+            }
+
+            private bool HasExcludedCodeAttribute<T>(T node, Func<T, SyntaxList<AttributeListSyntax>> getAttributeLists,
+                bool canBePartial = false)
+                where T : SyntaxNode
+            {
+                var hasExcludeFromCodeCoverageAttribute = getAttributeLists(node)
+                    .SelectMany(attributeList => attributeList.Attributes)
+                    .Any(HasExcludedAttribute);
+
+                if (!canBePartial)
```

Comment:
I think you could add `&& !node.Modifiers.HasAny(SyntaxKind.PartialKeyword)` (the code is from the top of my head, it should be changed) here to avoid getting the symbols for each method